usecases/auth/authentication/apikey/db_users.go

```diff
@@ -322,6 +358,12 @@ func (c *DBUser) storeToFile() error {
 	return os.Rename(tempFilename, c.path+"/"+FileName)
 }
 
+func (c *DBUser) Close() error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	return c.storeToFile()
```

Comment:
same here, bc you are doing the same operation in line 131, does it make sense then to put those locks into `storeToFile()` method?

Reply:
same as here https://github.com/weaviate/weaviate/pull/7786#discussion_r2041707194